src/driver/nativescript/NativescriptDriver.ts

```diff
@@ -0,0 +1,116 @@
+
+    constructor(connection: Connection) {
+        super(connection);
+
+        this.connection = connection;
+        this.options = connection.options as NativescriptConnectionOptions;
+        this.database = this.options.database;
+
+        // validate options to make sure everything is set
+        if (!this.options.database)
+            throw new DriverOptionNotSetError("database");
+
+        // load sqlite package
+        this.loadDependencies();
+    }
+
+
+    // -------------------------------------------------------------------------
+    // Public Methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Closes connection with database.
+     */
+    async disconnect(): Promise<void> {
+        return new Promise<void>((ok, fail) => {
+            this.queryRunner = undefined;
+            this.databaseConnection.close().then(ok).catch(fail);
+        });
+    }
+
+    /**
+     * Creates a query runner used to execute database queries.
+     */
+    createQueryRunner(mode: "master"|"slave" = "master"): QueryRunner {
+        if (!this.queryRunner)
+            this.queryRunner = new NativescriptQueryRunner(this);
+
+        return this.queryRunner;
+    }
+
+    normalizeType(column: { type?: ColumnType, length?: number | string, precision?: number|null, scale?: number }): string {
+        if ((column.type as any) === Buffer) {
+            return "blob";
+        }
+
+        return super.normalizeType(column);
+    }
+    // -------------------------------------------------------------------------
+    // Protected Methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Creates connection with the database.
+     */
+    protected createDatabaseConnection() {
+        return new Promise<void>((ok, fail) => {
+            const options = Object.assign({}, {
+                name: this.options.database,
+            }, this.options.extra || {});
+
+            new this.sqlite(options.name, (err: Error, db: any): any => {
+                if (err) return fail(err);
+
+                // use object mode to work with TypeORM
+                db.resultType(this.sqlite.RESULTSASOBJECT);
+
+
+                // we need to enable foreign keys in sqlite to make sure all foreign key related features
+                // working properly. this also makes onDelete work with sqlite.
+                db.execSQL(`PRAGMA foreign_keys = ON;`, [], (err: Error, result: any): any => {
+                    if (err) return fail(err);
+                    // We are all set
+                    ok(db);
+                });
+            });
+        });
+    }
+
+    /**
+     * If driver dependency is not given explicitly, then try to load it via "require".
+     */
+    protected loadDependencies(): void {
+        try {
+            this.sqlite = require("nativescript-sqlite");
```

Comment:
oh this should have been inside PlatformTools as all external dependencies (see #2531)

Reply:
I believe then, similar fix needs to be there for React Native itself too https://github.com/typeorm/typeorm/blob/master/src/driver/react-native/ReactNativeDriver.ts#L92

Reply:
The React Native problem is exactly the one I mentioned in #2240, but I didn't know enough about the codebase to figure out what needed to be fixed. I tried to fix by adding React Native support from within PlatformTools but encountered this error: https://github.com/typeorm/typeorm/blob/master/src/platform/BrowserPlatformTools.template#L33